jdaviz/configs/specviz2d/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -491,18 +580,47 @@ def create_bg_sub(self, add_data=False):
         return bg_sub_spec
 
     def vue_create_bg_sub(self, *args):
-        _ = self.create_bg_sub(add_data=True)
+        self.export_bg_sub(add_data=True)
 
     def _get_ext_trace(self):
-        return self.create_trace(add_data=False)
+        return self.export_trace(add_data=False)
 
     def _get_ext_input_spectrum(self):
         if self.ext_dataset_selected == 'From Plugin':
-            return self.create_bg_sub(add_data=False)
+            return self.export_bg_sub(add_data=False)
         else:
             return self.ext_dataset.selected_obj
 
-    def create_extract(self, add_data=False):
+    def import_extract(self, ext):
+        """
+        Import the input parameters from an existing specreduce extract object into the plugin.
+
+        Parameters
+        ----------
+        ext : specreduce.extract.BoxcarExtract
+            Extract object to import
+        """
+        if not isinstance(ext, extract.BoxcarExtract):
+            # TODO: add support for Optimal/Horne
+            raise TypeError("ext must be a specreduce.extract.BoxcarExtract object")
+
+        # TODO: should we detect/set the referenced dataset?
+        self.ext_width = ext.width
+
+    def export_extract(self, **kwargs):
+        """
+        Create a specreduce.extract object from the input parameters defined in the plugin.
```

Comment:
I'll need to think about this one.  The combination of this PR with #1572 will mean returning either BoxcarExtract or OptimalExtract.  So maybe we should:
* implement a base Extract class in specreduce?
* do the same as for trace and point to specreduce.extract module instead?

Reply:
I think the second option is better for now, alongside changing mentions of "[specreduce] Extract objects" in `docs/specviz2d/plugins.rst` to something more general like "[specreduce] extraction objects" until we implement something like your first bullet in `specreduce`.

Reply:
This line slipped through the cracks, but I modified the suggestion to match what we agreed to elsewhere. Since it's a small change, I will go ahead and give the approval so you can merge after you've made it.

Reply:
thanks, I guess I just fixed the ones in plugins.rst and forgot the one that you actually had the suggestion.  Fixed now.